src/input/accessors_impl.hh

```diff
@@ -240,7 +240,7 @@ template<class T>
 inline typename Iterator<T>::OutputType * Iterator<T>::operator ->() const {
     BOOST_STATIC_ASSERT(
```

Comment:
Add comment:
OutputType has to be an accessor to call its method, e.g.  iter->val<AnyType>("xyz"). iter has to be e.g. Iterator<Record>.